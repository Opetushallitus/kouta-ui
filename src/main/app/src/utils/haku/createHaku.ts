import { makeEntityMutator } from '#/src/utils/makeEntityMutator';

export const createHaku = makeEntityMutator(
  'put',
  'haku',
  'kouta-backend.haku'
);
