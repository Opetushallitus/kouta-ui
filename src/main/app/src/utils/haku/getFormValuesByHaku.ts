import { HakuFormValues } from '#/src/types/hakuTypes';
import { toSelectValue } from '#/src/utils';
import { getAjankohtaFields } from '#/src/utils/form/aloitusajankohtaHelpers';
import { getHakulomakeFieldsValues } from '#/src/utils/form/getHakulomakeFieldsValues';

export const getFormValuesByHaku = (haku): HakuFormValues => {
  const {
    muokkaaja,
    hakuajat = [],
    hakutapaKoodiUri = '',
    kohdejoukkoKoodiUri = '',
    kohdejoukonTarkenneKoodiUri = '',
    hakulomaketyyppi = '',
    hakulomakeAtaruId = '',
    hakulomakeKuvaus = {},
    hakulomakeLinkki = {},
    hakukohteenLiittamisenTakaraja,
    hakukohteenMuokkaamisenTakaraja,
    ajastettuJulkaisu,
    ajastettuHaunJaHakukohteidenArkistointi,
    kielivalinta = [],
    nimi = {},
    metadata = {},
    tila,
    externalId,
    organisaatioOid,
  } = haku;

  const {
    tulevaisuudenAikataulu = [],
    yhteyshenkilot = [],
    koulutuksenAlkamiskausi,
  } = metadata;

  return {
    organisaatioOid: toSelectValue(organisaatioOid),
    externalId,
    muokkaaja,
    tila,
    nimi,
    kieliversiot: kielivalinta,
    aikataulut: {
      ...getAjankohtaFields(koulutuksenAlkamiskausi),
      hakuaika: hakuajat,
      aikataulu: tulevaisuudenAikataulu,
      lisaamisenTakaraja: hakukohteenLiittamisenTakaraja,
      muokkauksenTakaraja: hakukohteenMuokkaamisenTakaraja,
      ajastettuJulkaisu,
      ajastettuHaunJaHakukohteidenArkistointi,
    },
    hakutapa: hakutapaKoodiUri,
    kohdejoukko: {
      kohdejoukko: kohdejoukkoKoodiUri,
      tarkenne: kohdejoukonTarkenneKoodiUri
        ? { value: kohdejoukonTarkenneKoodiUri }
        : null,
    },
    hakulomake: getHakulomakeFieldsValues({
      hakulomaketyyppi,
      hakulomakeAtaruId,
      hakulomakeKuvaus,
      hakulomakeLinkki,
    }),
    yhteyshenkilot: yhteyshenkilot.map(
      ({
        nimi,
        titteli,
        puhelinnumero,
        sahkoposti,
        wwwSivu,
        wwwSivuTeksti,
      }) => ({
        nimi: nimi || {},
        titteli: titteli || {},
        puhelinnumero: puhelinnumero || {},
        sahkoposti: sahkoposti || {},
        verkkosivu: wwwSivu || {},
        verkkosivuTeksti: wwwSivuTeksti || {},
      })
    ),
  };
};
