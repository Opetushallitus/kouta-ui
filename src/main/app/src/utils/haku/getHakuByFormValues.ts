import _fp from 'lodash/fp';

import { HakuFormValues } from '#/src/types/hakuTypes';
import { isPartialDate } from '#/src/utils';
import { getAlkamiskausiData } from '#/src/utils/form/aloitusajankohtaHelpers';
import { getHakulomakeFieldsData } from '#/src/utils/form/getHakulomakeFieldsData';
import isKorkeakoulutusKohdejoukkoKoodiUri from '#/src/utils/isKorkeakoulutusKohdejoukkoKoodiUri';

import {
  getKieleistyksetFromValues,
  getKielivalinta,
  getSerializedKieleistykset,
} from '../pickTranslations';

export const getHakuByFormValues = (values: HakuFormValues) => {
  const kielivalinta = getKielivalinta(values);

  const kieleistykset = getKieleistyksetFromValues(values);
  const kieleistyksetSerialized = getSerializedKieleistykset(values);
  const {
    hakulomaketyyppi,
    hakulomakeAtaruId,
    hakulomakeLinkki,
    hakulomakeKuvaus,
  } = getHakulomakeFieldsData({
    hakulomakeValues: values?.hakulomake,
    kieleistykset,
    kieleistyksetSerialized,
  });

  const kohdejoukkoKoodiUri = values?.kohdejoukko?.kohdejoukko || null;

  return {
    organisaatioOid: values?.organisaatioOid?.value,
    externalId: _fp.isEmpty(values?.externalId) ? null : values?.externalId,
    muokkaaja: values?.muokkaaja,
    tila: values?.tila,
    kielivalinta,
    hakutapaKoodiUri: values?.hakutapa || null,
    hakuajat: (values?.aikataulut?.hakuaika || []).map(
      ({ alkaa, paattyy }) => ({
        alkaa: isPartialDate(alkaa) ? null : alkaa,
        paattyy: isPartialDate(paattyy) ? null : paattyy,
      })
    ),
    hakukohteenLiittamisenTakaraja:
      values?.aikataulut?.lisaamisenTakaraja || null,
    nimi: kieleistykset(values?.nimi),
    kohdejoukkoKoodiUri,
    kohdejoukonTarkenneKoodiUri: isKorkeakoulutusKohdejoukkoKoodiUri(
      kohdejoukkoKoodiUri
    )
      ? values?.kohdejoukko?.tarkenne?.value || null
      : null,
    hakulomaketyyppi,
    metadata: {
      koulutuksenAlkamiskausi: getAlkamiskausiData(
        values?.aikataulut,
        kieleistyksetSerialized
      ),
      tulevaisuudenAikataulu: (values?.aikataulut?.aikataulu || []).map(
        ({ alkaa, paattyy }) => ({
          alkaa: alkaa || null,
          paattyy: paattyy || null,
        })
      ),
      yhteyshenkilot: (values?.yhteyshenkilot || []).map(
        ({
          nimi,
          titteli,
          puhelinnumero,
          sahkoposti,
          verkkosivu,
          verkkosivuTeksti,
        }) => ({
          nimi: kieleistykset(nimi),
          titteli: kieleistykset(titteli),
          puhelinnumero: kieleistykset(puhelinnumero),
          wwwSivu: kieleistykset(verkkosivu),
          wwwSivuTeksti: kieleistykset(verkkosivuTeksti),
          sahkoposti: kieleistykset(sahkoposti),
        })
      ),
    },
    hakukohteenMuokkaamisenTakaraja:
      values?.aikataulut?.muokkauksenTakaraja || null,
    hakukohteenLiittajaOrganisaatiot: values?.hakukohteenLiittajaOrganisaatiot,
    ajastettuJulkaisu: values?.aikataulut?.ajastettuJulkaisu || null,
    ajastettuHaunJaHakukohteidenArkistointi:
      values?.aikataulut?.ajastettuHaunJaHakukohteidenArkistointi || null,
    hakulomakeAtaruId,
    hakulomakeLinkki,
    hakulomakeKuvaus,
  };
};
