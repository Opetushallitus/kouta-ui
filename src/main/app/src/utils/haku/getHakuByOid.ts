import { ENTITY } from '#/src/constants';
import { useApiQuery } from '#/src/hooks/useApiQuery';
import _ from 'lodash';

export const getHakuByOid = async ({ oid, httpClient, apiUrls }) => {
  const { data, headers } = await httpClient.get(
    apiUrls.url('kouta-backend.haku-by-oid', oid)
  );

  const lastModified = _.get(headers, 'x-last-modified') || null;

  return _.isObject(data) ? { lastModified, ...data } : data;
};

export const useHakuByOid = (props, options = {}) =>
  useApiQuery(ENTITY.HAKU, props, getHakuByOid, {
    refetchOnWindowFocus: false,
    ...options,
  });

export default getHakuByOid;
