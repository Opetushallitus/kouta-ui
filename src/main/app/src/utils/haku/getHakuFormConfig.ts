import _fp from 'lodash/fp';
import { ifAny, otherwise } from '#/src/utils';
import {
  HAKULOMAKETYYPPI,
  JULKAISUTILA,
  TOTEUTUKSEN_AJANKOHTA,
} from '#/src/constants';
import isYhteishakuHakutapa from '#/src/utils/isYhteishakuHakutapa';
import isErillishakuHakutapa from '#/src/utils/isErillishakuHakutapa';
import createFormConfigBuilder from '#/src/utils/form/createFormConfigBuilder';
import {
  validateIfJulkaistu,
  getKielivalinta,
  kieliversiotSectionConfig,
  pohjaValintaSectionConfig,
  tilaSectionConfig,
  validateIf,
  validateOptionalTranslatedField,
} from '#/src/utils/form/formConfigUtils';
import {
  validateExistence,
  validateExistenceOfDate,
} from '#/src/utils/form/createErrorBuilder';

const getHakutapa = values => values?.hakutapa;

const config = createFormConfigBuilder().registerSections([
  kieliversiotSectionConfig,
  pohjaValintaSectionConfig,
  {
    section: 'nimi',
    parts: [
      {
        field: 'nimi',
        required: true,
        validate: (eb, values) =>
          eb.validateTranslations('nimi', getKielivalinta(values)),
      },
    ],
  },
  {
    section: 'kohdejoukko',
    parts: [
      {
        field: '.kohdejoukko',
        required: true,
        validate: validateIfJulkaistu(
          validateExistence('kohdejoukko.kohdejoukko')
        ),
      },
      {
        field: '.tarkenne',
      },
    ],
  },
  {
    section: 'hakutapa',
    parts: [
      {
        field: 'hakutapa',
        required: true,
        validate: validateIfJulkaistu(validateExistence('hakutapa')),
      },
    ],
  },
  {
    section: 'aikataulut',
    parts: [
      {
        field: '.hakuaikaGroup',
        required: true,
      },
      {
        field: '.toteutuksenAjankohta',
        validate: validateIfJulkaistu(
          validateExistence('aikataulut.toteutuksenAjankohta')
        ),
      },
      {
        field: '.kausi',
        required: values => isYhteishakuHakutapa(getHakutapa(values)),
      },
      {
        field: '.vuosi',
        validate: (eb, values) =>
          validateIf(
            isYhteishakuHakutapa(getHakutapa(values)) &&
              values?.aikataulut?.toteutuksenAjankohta ===
                TOTEUTUKSEN_AJANKOHTA.ALKAMISKAUSI &&
              values?.tila === JULKAISUTILA.JULKAISTU,
            _fp.pipe(
              validateExistence('aikataulut.kausi'),
              validateExistence('aikataulut.vuosi')
            )
          )(eb),
      },
      {
        field: '.tiedossaTarkkaAjankohta',
      },
      {
        field: '.tarkkaAlkaa',
        required: true,
        validate: (eb, values) =>
          validateIf(
            values?.aikataulut?.toteutuksenAjankohta ===
              TOTEUTUKSEN_AJANKOHTA.ALKAMISKAUSI &&
              values?.aikataulut?.tiedossaTarkkaAjankohta &&
              values?.tila === JULKAISUTILA.JULKAISTU,
            validateExistenceOfDate('aikataulut.tarkkaAlkaa')
          )(eb),
      },
      {
        field: '.tarkkaPaattyy',
      },
      {
        field: '.hakuaika',
        validate: validateIfJulkaistu((errorBuilder, values) =>
          errorBuilder
            .validateArrayMinLength('aikataulut.hakuaika', 1, {
              isFieldArray: true,
            })
            .validateArray('aikataulut.hakuaika', eb => {
              const hakutapa = getHakutapa(values);
              const isErillishaku = isErillishakuHakutapa(hakutapa);
              const isYhteishaku = isYhteishakuHakutapa(hakutapa);

              return _fp.flow([
                validateExistenceOfDate('alkaa'),
                validateIf(
                  isYhteishaku || isErillishaku,
                  validateExistenceOfDate('paattyy')
                ),
              ])(eb);
            })
        ),
      },
      {
        field: '.henkilokohtaisenSuunnitelmanLisatiedot',
        validate: validateOptionalTranslatedField(
          'aikataulut.henkilokohtaisenSuunnitelmanLisatiedot'
        ),
      },
      {
        field: '.hakuaika.alkaa',
        required: true,
      },
      {
        field: '.hakuaika.paattyy',
        required: values => {
          const hakutapa = getHakutapa(values);
          return (
            isYhteishakuHakutapa(hakutapa) || isErillishakuHakutapa(hakutapa)
          );
        },
      },
      {
        field: '.aikataulu',
      },
      {
        field: '.lisaamisenTakaraja',
      },
      {
        field: '.muokkauksenTakaraja',
      },
      {
        field: '.ajastettuJulkaisu',
      },
    ],
  },
  {
    section: 'hakulomake',
    parts: [
      {
        field: '.tyyppi',
        required: true,
        validate: validateIfJulkaistu(
          (eb, values) =>
            eb.validateExistence('hakulomake.tyyppi') &&
            _fp.cond([
              [
                ifAny(HAKULOMAKETYYPPI.ATARU),
                () => eb.validateExistence('hakulomake.lomake'),
              ],
              [
                ifAny(HAKULOMAKETYYPPI.MUU),
                () =>
                  eb.validateTranslations(
                    'hakulomake.linkki',
                    getKielivalinta(values)
                  ),
              ],
              [otherwise, () => eb],
            ])(tyyppi => values?.hakulomake?.tyyppi === tyyppi)
        ),
      },
      {
        field: '.lomake',
        required: true,
      },
      {
        field: '.linkki',
        required: true,
      },
      {
        field: '.kuvaus',
      },
    ],
  },
  {
    section: 'yhteyshenkilot',
    field: 'yhteyshenkilot',
  },
  tilaSectionConfig,
]);

const getHakuFormConfig = () => {
  return config.getConfig();
};

export default getHakuFormConfig;
