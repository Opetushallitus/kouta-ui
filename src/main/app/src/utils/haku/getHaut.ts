import { useApiQuery } from '#/src/hooks/useApiQuery';

const getHaut = async ({ organisaatioOid, httpClient, apiUrls, yhteishaut = true }) => {
  const { data } = await httpClient.get(
    apiUrls.url('kouta-backend.haku-list'),
    {
      params: { organisaatioOid, myosArkistoidut: true, yhteishaut },
    }
  );

  return data;
};

export const useHaut = (props, options = {}) =>
  useApiQuery('getHaut', getHaut, props, options);

export default getHaut;
