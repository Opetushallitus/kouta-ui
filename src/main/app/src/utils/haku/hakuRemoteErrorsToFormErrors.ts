import { RemoteErrorsToFormErrors } from '#/src/types/formTypes';

export const hakuRemoteErrorsToFormErrors: RemoteErrorsToFormErrors = ({
  errorType,
  msg,
  path,
}) => {
  if (path === 'tila') {
    if (errorType === 'integrityViolation' && /hakukohteita/i.test(msg)) {
      return {
        field: 'tila',
        errorKey:
          'ilmoitukset.poistoEstettyRiippuvuuksienTakia.haullaHakukohteita',
      };
    }
  }
};
