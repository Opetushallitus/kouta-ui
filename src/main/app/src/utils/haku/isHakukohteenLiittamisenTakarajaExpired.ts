const isHakukohteenLiittamisenTakarajaExpired = (now, liittamisenTakaraja) => {
  return now >= liittamisenTakaraja;
};

export default isHakukohteenLiittamisenTakarajaExpired;
