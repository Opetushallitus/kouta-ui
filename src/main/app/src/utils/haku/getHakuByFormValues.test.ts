import { mapValues, merge } from 'lodash';

import { parseEditorState } from '#/src/components/Editor/utils';
import {
  HAKULOMAKETYYPPI,
  JULKAISUTILA,
  Alkamiskausityyppi,
} from '#/src/constants';
import { HakuFormValues } from '#/src/types/hakuTypes';
import { getHakuByFormValues } from '#/src/utils/haku/getHakuByFormValues';

const baseValues: HakuFormValues = {
  muokkaaja: '1.1.1.1',
  tila: JULKAISUTILA.TALLENNETTU,
  nimi: {
    fi: 'Nimi',
    sv: 'Namn',
  },
  kieliversiot: ['fi', 'sv'],
  aikataulut: {
    ajankohtaTyyppi: Alkamiskausityyppi.ALKAMISKAUSI_JA_VUOSI,
    kausi: 'alkamiskausi_1#1',
    vuosi: { value: '2020' },
    hakuaika: [
      {
        alkaa: '2019-04-16T08:44',
        paattyy: '2019-04-18T08:44',
      },
      {
        alkaa: '2019-06-16T08:44',
        paattyy: '2019-06-18T08:44',
      },
    ],
    aikataulu: [
      {
        alkaa: '2019-08-16T08:44',
        paattyy: '2019-08-18T08:44',
      },
      {
        alkaa: '2019-09-16T08:44',
        paattyy: '2019-09-18T08:44',
      },
    ],
    lisaamisenTakaraja: '2019-09-16T08:44',
    muokkauksenTakaraja: '2019-10-16T08:44',
    ajastettuJulkaisu: '2019-11-16T08:44',
  },
  hakutapa: 'hakutapa_1#1',
  kohdejoukko: {
    kohdejoukko: 'haunkohdejoukko_12',
    tarkenne: { value: 'tarkenne_1#1' },
  },
  hakulomake: {
    tyyppi: HAKULOMAKETYYPPI.ATARU,
    lomake: { value: '12345' },
  },
  yhteyshenkilot: [
    {
      nimi: {
        fi: 'Fi nimi',
        sv: 'Sv nimi',
      },
      titteli: { fi: 'Fi titteli', sv: 'Sv titteli' },
      sahkoposti: { fi: 'Fi sähköposti', sv: 'Sv sähköposti' },
      puhelinnumero: { fi: 'Fi puhelinnumero', sv: 'Sv puhelinnumero' },
      verkkosivu: { fi: 'Fi verkkosivu', sv: 'Sv verkkosivu' },
    },
  ],
};

test('getHakuByFormValues returns correct haku given form values', () => {
  const haku = getHakuByFormValues(baseValues);

  expect(haku).toMatchSnapshot();
});

test('getHakuByFormValues returns correct haku given different hakulomake variations', () => {
  const hakuMuu = getHakuByFormValues(
    merge({}, baseValues, {
      hakulomake: {
        tyyppi: HAKULOMAKETYYPPI.MUU,
        linkki: { fi: 'https://google.fi' },
      },
    })
  );

  const hakuEiHakua = getHakuByFormValues(
    merge({}, baseValues, {
      hakulomake: {
        tyyppi: HAKULOMAKETYYPPI.EI_SAHKOISTA_HAKUA,
        kuvaus: { fi: parseEditorState('kuvaus') },
      },
    })
  );

  expect(hakuMuu).toMatchSnapshot();
  expect(hakuEiHakua).toMatchSnapshot();
});

test('getHakuByFormValues toteutuksen ajankohta - Tarkka alkamisaika', () => {
  expect(
    getHakuByFormValues(
      merge({}, baseValues, {
        aikataulut: {
          ajankohtaTyyppi: Alkamiskausityyppi.TARKKA_ALKAMISAJANKOHTA,
          tarkkaAlkaa: '2019-09-16T08:44',
          tarkkaPaattyy: '2019-09-16T08:44',
        },
      })
    )
  ).toMatchSnapshot();
});

test('getHakuByFormValues toteutuksen ajankohta - Aloitus henkilokohtaisen suunnitelman mukaisesti', () => {
  expect(
    getHakuByFormValues(
      merge({}, baseValues, {
        aikataulut: {
          ajankohtaTyyppi: Alkamiskausityyppi.HENKILOKOHTAINEN_SUUNNITELMA,
          henkilokohtaisenSuunnitelmanLisatiedot: mapValues(
            {
              fi: '<p>hlokoht fi </p>',
              sv: '<p>hlokoht sv </p>',
              en: '<p>hlokoht en </p>',
            },
            parseEditorState
          ),
        },
      })
    )
  ).toMatchSnapshot();
});
