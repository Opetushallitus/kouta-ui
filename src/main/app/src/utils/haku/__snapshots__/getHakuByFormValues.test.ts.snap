// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getHakuByFormValues returns correct haku given different hakulomake variations 1`] = `
{
  "ajastettuHaunJaHakukohteidenArkistointi": "2019-11-16T09:44",
  "ajastettuJulkaisu": "2019-11-16T08:44",
  "externalId": "ext1",
  "hakuajat": [
    {
      "alkaa": "2019-04-16T08:44",
      "paattyy": "2019-04-18T08:44",
    },
    {
      "alkaa": "2019-06-16T08:44",
      "paattyy": "2019-06-18T08:44",
    },
  ],
  "hakukohteenLiittamisenTakaraja": "2019-09-16T08:44",
  "hakukohteenMuokkaamisenTakaraja": "2019-10-16T08:44",
  "hakulomakeAtaruId": null,
  "hakulomakeKuvaus": {},
  "hakulomakeLinkki": {
    "fi": "https://google.fi",
  },
  "hakulomaketyyppi": "muu",
  "hakutapaKoodiUri": "hakutapa_1#1",
  "kielivalinta": [
    "fi",
    "sv",
  ],
  "kohdejoukkoKoodiUri": "haunkohdejoukko_12",
  "kohdejoukonTarkenneKoodiUri": "tarkenne_1#1",
  "metadata": {
    "koulutuksenAlkamiskausi": {
      "alkamiskausityyppi": "alkamiskausi ja -vuosi",
      "henkilokohtaisenSuunnitelmanLisatiedot": {},
      "koulutuksenAlkamiskausiKoodiUri": "alkamiskausi_1#1",
      "koulutuksenAlkamispaivamaara": null,
      "koulutuksenAlkamisvuosi": 2020,
      "koulutuksenPaattymispaivamaara": null,
    },
    "tulevaisuudenAikataulu": [
      {
        "alkaa": "2019-08-16T08:44",
        "paattyy": "2019-08-18T08:44",
      },
      {
        "alkaa": "2019-09-16T08:44",
        "paattyy": "2019-09-18T08:44",
      },
    ],
    "yhteyshenkilot": [
      {
        "nimi": {
          "fi": "Fi nimi",
          "sv": "Sv nimi",
        },
        "puhelinnumero": {
          "fi": "Fi puhelinnumero",
          "sv": "Sv puhelinnumero",
        },
        "sahkoposti": {
          "fi": "Fi sähköposti",
          "sv": "Sv sähköposti",
        },
        "titteli": {
          "fi": "Fi titteli",
          "sv": "Sv titteli",
        },
        "wwwSivu": {
          "fi": "Fi verkkosivu",
          "sv": "Sv verkkosivu",
        },
        "wwwSivuTeksti": {},
      },
    ],
  },
  "muokkaaja": "1.1.1.1",
  "nimi": {
    "fi": "Nimi",
    "sv": "Namn",
  },
  "organisaatioOid": undefined,
  "tila": "tallennettu",
}
`;

exports[`getHakuByFormValues returns correct haku given different hakulomake variations 2`] = `
{
  "ajastettuHaunJaHakukohteidenArkistointi": "2019-11-16T09:44",
  "ajastettuJulkaisu": "2019-11-16T08:44",
  "externalId": "ext1",
  "hakuajat": [
    {
      "alkaa": "2019-04-16T08:44",
      "paattyy": "2019-04-18T08:44",
    },
    {
      "alkaa": "2019-06-16T08:44",
      "paattyy": "2019-06-18T08:44",
    },
  ],
  "hakukohteenLiittamisenTakaraja": "2019-09-16T08:44",
  "hakukohteenMuokkaamisenTakaraja": "2019-10-16T08:44",
  "hakulomakeAtaruId": null,
  "hakulomakeKuvaus": {
    "fi": "<p>kuvaus</p>",
  },
  "hakulomakeLinkki": {},
  "hakulomaketyyppi": "ei sähköistä",
  "hakutapaKoodiUri": "hakutapa_1#1",
  "kielivalinta": [
    "fi",
    "sv",
  ],
  "kohdejoukkoKoodiUri": "haunkohdejoukko_12",
  "kohdejoukonTarkenneKoodiUri": "tarkenne_1#1",
  "metadata": {
    "koulutuksenAlkamiskausi": {
      "alkamiskausityyppi": "alkamiskausi ja -vuosi",
      "henkilokohtaisenSuunnitelmanLisatiedot": {},
      "koulutuksenAlkamiskausiKoodiUri": "alkamiskausi_1#1",
      "koulutuksenAlkamispaivamaara": null,
      "koulutuksenAlkamisvuosi": 2020,
      "koulutuksenPaattymispaivamaara": null,
    },
    "tulevaisuudenAikataulu": [
      {
        "alkaa": "2019-08-16T08:44",
        "paattyy": "2019-08-18T08:44",
      },
      {
        "alkaa": "2019-09-16T08:44",
        "paattyy": "2019-09-18T08:44",
      },
    ],
    "yhteyshenkilot": [
      {
        "nimi": {
          "fi": "Fi nimi",
          "sv": "Sv nimi",
        },
        "puhelinnumero": {
          "fi": "Fi puhelinnumero",
          "sv": "Sv puhelinnumero",
        },
        "sahkoposti": {
          "fi": "Fi sähköposti",
          "sv": "Sv sähköposti",
        },
        "titteli": {
          "fi": "Fi titteli",
          "sv": "Sv titteli",
        },
        "wwwSivu": {
          "fi": "Fi verkkosivu",
          "sv": "Sv verkkosivu",
        },
        "wwwSivuTeksti": {},
      },
    ],
  },
  "muokkaaja": "1.1.1.1",
  "nimi": {
    "fi": "Nimi",
    "sv": "Namn",
  },
  "organisaatioOid": undefined,
  "tila": "tallennettu",
}
`;

exports[`getHakuByFormValues returns correct haku given form values 1`] = `
{
  "ajastettuHaunJaHakukohteidenArkistointi": "2019-11-16T09:44",
  "ajastettuJulkaisu": "2019-11-16T08:44",
  "externalId": "ext1",
  "hakuajat": [
    {
      "alkaa": "2019-04-16T08:44",
      "paattyy": "2019-04-18T08:44",
    },
    {
      "alkaa": "2019-06-16T08:44",
      "paattyy": "2019-06-18T08:44",
    },
  ],
  "hakukohteenLiittamisenTakaraja": "2019-09-16T08:44",
  "hakukohteenMuokkaamisenTakaraja": "2019-10-16T08:44",
  "hakulomakeAtaruId": "12345",
  "hakulomakeKuvaus": {},
  "hakulomakeLinkki": {},
  "hakulomaketyyppi": "ataru",
  "hakutapaKoodiUri": "hakutapa_1#1",
  "kielivalinta": [
    "fi",
    "sv",
  ],
  "kohdejoukkoKoodiUri": "haunkohdejoukko_12",
  "kohdejoukonTarkenneKoodiUri": "tarkenne_1#1",
  "metadata": {
    "koulutuksenAlkamiskausi": {
      "alkamiskausityyppi": "alkamiskausi ja -vuosi",
      "henkilokohtaisenSuunnitelmanLisatiedot": {},
      "koulutuksenAlkamiskausiKoodiUri": "alkamiskausi_1#1",
      "koulutuksenAlkamispaivamaara": null,
      "koulutuksenAlkamisvuosi": 2020,
      "koulutuksenPaattymispaivamaara": null,
    },
    "tulevaisuudenAikataulu": [
      {
        "alkaa": "2019-08-16T08:44",
        "paattyy": "2019-08-18T08:44",
      },
      {
        "alkaa": "2019-09-16T08:44",
        "paattyy": "2019-09-18T08:44",
      },
    ],
    "yhteyshenkilot": [
      {
        "nimi": {
          "fi": "Fi nimi",
          "sv": "Sv nimi",
        },
        "puhelinnumero": {
          "fi": "Fi puhelinnumero",
          "sv": "Sv puhelinnumero",
        },
        "sahkoposti": {
          "fi": "Fi sähköposti",
          "sv": "Sv sähköposti",
        },
        "titteli": {
          "fi": "Fi titteli",
          "sv": "Sv titteli",
        },
        "wwwSivu": {
          "fi": "Fi verkkosivu",
          "sv": "Sv verkkosivu",
        },
        "wwwSivuTeksti": {},
      },
    ],
  },
  "muokkaaja": "1.1.1.1",
  "nimi": {
    "fi": "Nimi",
    "sv": "Namn",
  },
  "organisaatioOid": undefined,
  "tila": "tallennettu",
}
`;

exports[`getHakuByFormValues toteutuksen ajankohta - Aloitus henkilokohtaisen suunnitelman mukaisesti 1`] = `
{
  "ajastettuHaunJaHakukohteidenArkistointi": "2019-11-16T09:44",
  "ajastettuJulkaisu": "2019-11-16T08:44",
  "externalId": "ext1",
  "hakuajat": [
    {
      "alkaa": "2019-04-16T08:44",
      "paattyy": "2019-04-18T08:44",
    },
    {
      "alkaa": "2019-06-16T08:44",
      "paattyy": "2019-06-18T08:44",
    },
  ],
  "hakukohteenLiittamisenTakaraja": "2019-09-16T08:44",
  "hakukohteenMuokkaamisenTakaraja": "2019-10-16T08:44",
  "hakulomakeAtaruId": "12345",
  "hakulomakeKuvaus": {},
  "hakulomakeLinkki": {},
  "hakulomaketyyppi": "ataru",
  "hakutapaKoodiUri": "hakutapa_1#1",
  "kielivalinta": [
    "fi",
    "sv",
  ],
  "kohdejoukkoKoodiUri": "haunkohdejoukko_12",
  "kohdejoukonTarkenneKoodiUri": "tarkenne_1#1",
  "metadata": {
    "koulutuksenAlkamiskausi": {
      "alkamiskausityyppi": "henkilokohtainen suunnitelma",
      "henkilokohtaisenSuunnitelmanLisatiedot": {
        "fi": "<p>hlokoht fi </p>",
        "sv": "<p>hlokoht sv </p>",
      },
      "koulutuksenAlkamiskausiKoodiUri": "alkamiskausi_1#1",
      "koulutuksenAlkamispaivamaara": null,
      "koulutuksenAlkamisvuosi": 2020,
      "koulutuksenPaattymispaivamaara": null,
    },
    "tulevaisuudenAikataulu": [
      {
        "alkaa": "2019-08-16T08:44",
        "paattyy": "2019-08-18T08:44",
      },
      {
        "alkaa": "2019-09-16T08:44",
        "paattyy": "2019-09-18T08:44",
      },
    ],
    "yhteyshenkilot": [
      {
        "nimi": {
          "fi": "Fi nimi",
          "sv": "Sv nimi",
        },
        "puhelinnumero": {
          "fi": "Fi puhelinnumero",
          "sv": "Sv puhelinnumero",
        },
        "sahkoposti": {
          "fi": "Fi sähköposti",
          "sv": "Sv sähköposti",
        },
        "titteli": {
          "fi": "Fi titteli",
          "sv": "Sv titteli",
        },
        "wwwSivu": {
          "fi": "Fi verkkosivu",
          "sv": "Sv verkkosivu",
        },
        "wwwSivuTeksti": {},
      },
    ],
  },
  "muokkaaja": "1.1.1.1",
  "nimi": {
    "fi": "Nimi",
    "sv": "Namn",
  },
  "organisaatioOid": undefined,
  "tila": "tallennettu",
}
`;

exports[`getHakuByFormValues toteutuksen ajankohta - Tarkka alkamisaika 1`] = `
{
  "ajastettuHaunJaHakukohteidenArkistointi": "2019-11-16T09:44",
  "ajastettuJulkaisu": "2019-11-16T08:44",
  "externalId": "ext1",
  "hakuajat": [
    {
      "alkaa": "2019-04-16T08:44",
      "paattyy": "2019-04-18T08:44",
    },
    {
      "alkaa": "2019-06-16T08:44",
      "paattyy": "2019-06-18T08:44",
    },
  ],
  "hakukohteenLiittamisenTakaraja": "2019-09-16T08:44",
  "hakukohteenMuokkaamisenTakaraja": "2019-10-16T08:44",
  "hakulomakeAtaruId": "12345",
  "hakulomakeKuvaus": {},
  "hakulomakeLinkki": {},
  "hakulomaketyyppi": "ataru",
  "hakutapaKoodiUri": "hakutapa_1#1",
  "kielivalinta": [
    "fi",
    "sv",
  ],
  "kohdejoukkoKoodiUri": "haunkohdejoukko_12",
  "kohdejoukonTarkenneKoodiUri": "tarkenne_1#1",
  "metadata": {
    "koulutuksenAlkamiskausi": {
      "alkamiskausityyppi": "tarkka alkamisajankohta",
      "henkilokohtaisenSuunnitelmanLisatiedot": {},
      "koulutuksenAlkamiskausiKoodiUri": "alkamiskausi_1#1",
      "koulutuksenAlkamispaivamaara": "2019-09-16T08:44",
      "koulutuksenAlkamisvuosi": 2020,
      "koulutuksenPaattymispaivamaara": "2019-09-16T08:44",
    },
    "tulevaisuudenAikataulu": [
      {
        "alkaa": "2019-08-16T08:44",
        "paattyy": "2019-08-18T08:44",
      },
      {
        "alkaa": "2019-09-16T08:44",
        "paattyy": "2019-09-18T08:44",
      },
    ],
    "yhteyshenkilot": [
      {
        "nimi": {
          "fi": "Fi nimi",
          "sv": "Sv nimi",
        },
        "puhelinnumero": {
          "fi": "Fi puhelinnumero",
          "sv": "Sv puhelinnumero",
        },
        "sahkoposti": {
          "fi": "Fi sähköposti",
          "sv": "Sv sähköposti",
        },
        "titteli": {
          "fi": "Fi titteli",
          "sv": "Sv titteli",
        },
        "wwwSivu": {
          "fi": "Fi verkkosivu",
          "sv": "Sv verkkosivu",
        },
        "wwwSivuTeksti": {},
      },
    ],
  },
  "muokkaaja": "1.1.1.1",
  "nimi": {
    "fi": "Nimi",
    "sv": "Namn",
  },
  "organisaatioOid": undefined,
  "tila": "tallennettu",
}
`;
